htmlelements-java/src/main/java/ru/yandex/qatools/htmlelements/element/Radio.java

```diff
@@ -49,49 +50,39 @@ public Radio(WebElement wrappedElement) {
      * @return {@code WebElement} representing selected radio button or {@code null} if no radio buttons are selected.
      */
     public WebElement getSelectedButton() {
-        for (WebElement button : getButtons()) {
-            if (button.isSelected()) {
-                return button;
-            }
-        }
-
-        throw new NoSuchElementException("No selected button");
+        return getButtons().stream()
+                .filter(WebElement::isSelected)
+                .findAny()
+                .orElseThrow(() -> new NoSuchElementException("No selected button"));
     }
 
     /**
-     * Indicates if radio has selected button.
+     * Indicates if radio group has selected button.
      *
      * @return {@code true} if radio has selected button and {@code false} otherwise.
      */
     public boolean hasSelectedButton() {
-        for (WebElement button : getButtons()) {
-            if (button.isSelected()) {
-                return true;
-            }
-        }
-
-        return false;
+        return getButtons().stream()
+                .anyMatch(WebElement::isSelected);
     }
 
     /**
-     * Selects radio button that have a value matching the specified argument.
+     * Selects first radio button that has a value matching the specified argument.
      *
      * @param value The value to match against.
      */
     public void selectByValue(String value) {
-        for (WebElement button : getButtons()) {
-            String buttonValue = button.getAttribute("value");
-            if (value.equals(buttonValue)) {
-                selectButton(button);
-                return;
-            }
-        }
+        Optional<WebElement> matchingButton = getButtons().stream()
+                .filter(b -> b.getAttribute("value").equals(value))
```

Comment:
not NPE safe

should be `value.equals(buttonValue)` with precondition check, or `StringUtils.equals(value, btnVal)` if commons-lang3 available in classpath